src/libmscore/draw/drawcomp.cpp

```diff
@@ -0,0 +1,443 @@
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "drawcomp.h"
+
+#include "realfn.h"
+#include "log.h"
+
+using namespace mu;
+using namespace mu::draw;
+
+namespace mu::draw::comp {
+static const int DEFAULT_PREC(3);
+
+template<class T>
+static bool isEqual(const std::vector<T>& v1, const std::vector<T>& v2, DrawComp::Tolerance tolerance);
+
+static bool isEqual(const qreal& v1, const qreal& v2, double tolerance)
+{
+    if (tolerance > 0) {
+        double delta = RealFloor(v1, DEFAULT_PREC) - RealFloor(v2, DEFAULT_PREC);
+        return std::fabs(delta) < tolerance;
+    }
+    return RealIsEqual(RealFloor(v1, DEFAULT_PREC), RealFloor(v2, DEFAULT_PREC));
+}
+
+static bool isEqual(const QPointF& p1, const QPointF& p2, double tolerance)
+{
+    if (!isEqual(p1.x(), p2.x(), tolerance)) {
+        return false;
+    }
+
+    if (!isEqual(p1.y(), p2.y(), tolerance)) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool isEqual(const QRectF& r1, const QRectF& r2, double tolerance)
+{
+    if (!isEqual(r1.x(), r2.x(), tolerance)) {
+        return false;
+    }
+
+    if (!isEqual(r1.y(), r2.y(), tolerance)) {
+        return false;
+    }
+
+    if (!isEqual(r1.width(), r2.width(), tolerance)) {
+        return false;
+    }
+
+    if (!isEqual(r1.height(), r2.height(), tolerance)) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool isEqual(const QSize& sz1, const QSize& sz2)
+{
+    if (sz1.width() != sz2.width()) {
+        return false;
+    }
+
+    if (sz1.height() != sz2.height()) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool isEqual(const QPen& p1, const QPen& p2)
+{
+    if (p1.style() != p2.style()) {
+        return false;
+    }
+
+    if (p1.width() != p2.width()) {
+        return false;
+    }
+
+    if (p1.color().name() != p2.color().name()) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool isEqual(const QBrush& b1, const QBrush& b2)
+{
+    if (b1.style() != b2.style()) {
+        return false;
+    }
+
+    if (b1.color().name() != b2.color().name()) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool isEqual(const QFont& f1, const QFont& f2)
+{
+    if (f1.family() != f2.family()) {
+        return false;
+    }
+
+    if (f1.pointSize() != f2.pointSize()) {
+        return false;
+    }
+
+    if (f1.weight() != f2.weight()) {
+        return false;
+    }
+
+    if (f1.italic() != f2.italic()) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool isEqual(const QTransform& t1, const QTransform& t2)
+{
+    if (t1 != t2) {
```

Comment:
You can simplify this function:

 static bool isEqual(const QTransform& t1, const QTransform& t2) {
 return t1 == t2;
 }

Reply:
I want all methods to be consistent